k8s-operator/sessionrecording/hijacker.go

```diff
@@ -174,7 +191,6 @@ func (h *Hijacker) setUpRecording(ctx context.Context, conn net.Conn) (net.Conn,
 		}
 		msg += "; failure mode set to 'fail closed'; closing connection"
 		h.log.Error(msg)
-		lc.Fail()
```

Comment:
why remove this?

Reply:
We were previously writing any bytes leftover in buffer to the connection on close, but not if the connection had failed and this method was setting state to failed. It didnt' seem like in practice there is any case where there are such leftover bytes, so this PR removes the writing of them, so there is no more need to set the state to failed as nothing else needs to know that

Reply:
https://github.com/tailscale/tailscale/pull/12947/commits/5bbf826430f4e390906a848cd51cbceb63e2863c#diff-ab30d85e114d574643cfe3032c3fa880f4ec1226cb84eeb716b7b4f5394fc4acL175-L177